receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/accumulator.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecscontainermetrics
+
+import (
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+)
+
+// metricDataAccumulator defines the accumulator
+type metricDataAccumulator struct {
+	md []*consumerdata.MetricsData
+}
+
+// getMetricsData generates OT Metrics data from task metadata and docker stats
+func (acc *metricDataAccumulator) getMetricsData(containerStatsMap map[string]ContainerStats, metadata TaskMetadata) {
+
+	taskMetrics := ECSMetrics{}
+	timestamp := timestampProto(time.Now())
+	taskResources := taskResources(metadata)
+
+	for _, containerMetadata := range metadata.Containers {
+		stats := containerStatsMap[containerMetadata.DockerID]
+		containerMetrics := getContainerMetrics(stats)
+		containerMetrics.MemoryReserved = *containerMetadata.Limits.Memory
+		containerMetrics.CPUReserved = *containerMetadata.Limits.CPU
+
+		containerResources := containerResources(containerMetadata)
+		for k, v := range taskResources.Labels {
+			containerResources.Labels[k] = v
+		}
+
+		acc.accumulate(
+			containerResources,
+			convertToOCMetrics(ContainerPrefix, containerMetrics, nil, nil, timestamp),
+		)
+
+		aggregateTaskMetrics(&taskMetrics, containerMetrics)
+	}
+
+	// Overwrite Memory limit with task level limit
+	if metadata.Limits.Memory != nil {
+		taskMetrics.MemoryReserved = *metadata.Limits.Memory
+	}
+
+	taskMetrics.CPUReserved = taskMetrics.CPUReserved / CPUsInVCpu
+
+	// Overwrite CPU limit with task level limit
+	if metadata.Limits.CPU != nil {
+		taskMetrics.CPUReserved = *metadata.Limits.CPU
+	}
+
+	acc.accumulate(
+		taskResources,
+		convertToOCMetrics(TaskPrefix, taskMetrics, nil, nil, timestamp),
```

Comment:
If the 3rd and 4th parameters to this method are always nil, I would remove those parameters.

Reply:
I also thought about it while writing this piece of code. Here, I kept the skeleton ready and the same method can be utilized to set metric labels. In our next PRs, we can just pass the `LabelKeys` and `LabelValues` and we are done. If we really don't utilize, I will remove them.

Reply:
👍 SGTM